src/sentry/services/hybrid_cloud/app/impl.py

```diff
@@ -1,18 +1,32 @@
 from __future__ import annotations
 
-from typing import List
+import datetime
+from typing import List, Optional
 
+from sentry.constants import SentryAppInstallationStatus
 from sentry.models import SentryApp, SentryAppInstallation
 from sentry.services.hybrid_cloud.app import ApiSentryAppInstallation, AppService
 
 
 class DatabaseBackedAppService(AppService):
-    def get_installed_for_organization(
-        self, *, organization_id: int
+    def get_many(
+        self,
+        *,
+        organization_id: Optional[int] = None,
+        status: Optional[int] = SentryAppInstallationStatus.INSTALLED,
+        date_deleted: Optional[datetime.datetime] = None,
+        api_token_id: Optional[int] = None,
     ) -> List[ApiSentryAppInstallation]:
-        installations = SentryAppInstallation.objects.get_installed_for_organization(
-            organization_id
-        ).select_related("sentry_app")
+        installations = SentryAppInstallation.objects.select_related("sentry_app")
+        if organization_id is not None:
+            installations = installations.filter(
+                organization_id=organization_id,
+                date_deleted=date_deleted,
```

Comment:
Do we need to apply  the `date_deleted` condition on its own ever?

Reply:
I didn't haven't seen any cases of that yet. It seems like it's often used to exclude deleted apps. We could perhaps just simplify this interface.

Reply:
I, honestly think I'm going to reverse this change, and have two simple interfaces: `get_installed_for_organization` and `get_for_api_token`.  The other filtering can be done by hand.  

I sincerely worry for security bugs where we return all app installation obects.  I also worry this interface prevents any lateral caching strategies.

Reply:
Apologies, my comment was in the wrong place.  Anyways, it'll be more clear after my commit.